bin/rpm_filter.py

```diff
@@ -0,0 +1,214 @@
+    df.iloc[:, reads_col_index] = df.iloc[:, reads_col_index].astype(int)
+    # Assign new column names if the user has given them otherwise go with the already
+    # existing ones except rename the reads column
+    if col_names:
+        df.columns = col_names
+    else:
+        df.columns.values[reads_col_index] = reads_col_name
+    return df
+
+
+def get_sum_of_column(df: pd.DataFrame, column_to_get_sum_of: int) -> int:
+    """Get the sum of a given column
+
+    Args:
+        df (pd.DataFramem): DataFrame where the column is
+        column_to_get_sum_of (int): The 0-based index of the column to get_sum of
+
+    Returns:
+        int: The sum of the column
+    """
+    col_sum: int = df.iloc[:, column_to_get_sum_of].sum()
+    logger.info("The sum of the given column is: %s", str(col_sum))
+    return col_sum
+
+
+def add_rpm_column(
+    df: pd.DataFrame, col_sum_per_million: int, col_name: str = "reads_count"
+) -> pd.DataFrame:
+    """Add Reads per million column to given data frame
+
+    Args:
+        df (pd.DataFrame): Pandas DataFrame where the RPM column is to be added
+        col_sum_per_million (int): Sum of all reads divided by million
+        col_name (str, optional): Column name where the reads counts assigned to taxons are. Defaults to "reads_count".
+
+    Returns:
+        pd.DataFrame: DataFrame where the RPM column is added
+    """
+    df["RPM"] = df.apply(lambda row: row[col_name] / col_sum_per_million, axis=1)
+    df["RPM"] = df.apply(lambda row: round(row["RPM"], 1), axis=1)
+    return df
+
+
+def filter_by_rpm(df: pd.DataFrame, rpm_value: float) -> pd.DataFrame:
+    """Filter rows where the RPM value is higher than the given value
+
+    Args:
+        df (pd.DataFrame): The DataFrame where to filter by given RPM value
+        rpm_value (float): The threshold RPM value to use for filtering
+
+    Returns:
+        pd.DataFrame: The DataFrame where the RPM filtering has been applied
+    """
+    return df[df["RPM"] >= rpm_value]
+
+
+def post_process_df(df: pd.DataFrame, index_name: str = "line_number") -> pd.DataFrame:
+    """Drop old index and give the new index a name
+
+    Args:
+        df (pd.DataFrame): DataFrame where the post processing is to be done
+        index_name (str, optional): The new name for the index. Defaults to "line_number".
+
+    Returns:
+        pd.DataFrame: Transformed DataFrame where index is updated
+    """
+    df = df.reset_index(drop=True)
+    df.index.name = index_name
+    return df
+
+
+def main(argv=None):
+    """Coordinate argument parsing and program execution."""
+    args = parse_args(argv)
+    logging.basicConfig(level=args.log_level, format="[%(levelname)s] %(message)s")
+    classifier_file: Path = args.classifier_output_file
+    if not classifier_file.is_file():
+        logger.error("The given input file %s was not found!", classifier_file)
+        sys.exit(1)
+    # Should some column name be read as string?
+    if args.string_colname:
```

Comment:
In the kaiju output files e.g. `taxon_id` could be read as string instead of integer. This enforces that that column is read as string when importing it as a pandas dataframe.